test/StreamJsonRpc.Tests/MarshalableProxyTests.cs

```diff
@@ -728,4 +793,44 @@ public void Dispose()
         {
         }
     }
+
+    public class MarshalableWithSubTypes : IMarshalableWithSubTypes
+    {
+        public Task<int> GetAsync(int value) => Task.FromResult(value);
+
+        public void Dispose()
+        {
+        }
+    }
+
+    public class MarshalableSubType1 : IMarshalableSubType1
+    {
+        public Task<int> GetAsync(int value) => Task.FromResult(value);
+
+        public Task<int> GetPlusOneAsync(int value) => Task.FromResult(value + 1);
+
+        public void Dispose()
+        {
+        }
+    }
+
+    public class MarshalableSubType2 : IMarshalableSubType2
```

Comment:
_One_ of these test classes should derive from another _class_ that implements the interface, to ensure we're not looking only one-level deep for interface implementations.